### ISSUE TITLE
[BUG]: Mojo 🔥 v0.5.0 VSCode extension not reporting compiler error

### ISSUE THREAD

**Issue by isuckatcs** [modular/mojo#1348]
### Bug description

Mutating an immutable variable is not allowed in mojo, so the compiler reports an error, which is detected by the VSCode mojo addon and is reported inside the editor too.

```mojo
fn main():
    let x = 3
    x = 4
```
![error reported](https://github.com/modularml/mojo/assets/65320245/95b44d74-e948-4b81-8a0b-5007188bc855)

In case when the variable is not initialized upon declaration, the same error is not reported inside VSCode, even tough the compiler still detects it.
```mojo
fn main():
    let z: Int
    z = 3
    z = 4
```
```
> mojo bug.🔥 
.../bug.🔥:4:9: error: invalid mutation of immutable value 'z.value'
    z = 4
        ^
.../bug.🔥:2:5: note: 'z' declared here
    let z: Int
    ^
```
![error not reported](https://github.com/modularml/mojo/assets/65320245/3ab34311-eade-4df2-8ed2-69c3c706a80c)

### Steps to reproduce

Open a mojo source file with the following content in VSCode with the Mojo addon installed.
```mojo
fn main():
    let z: Int
    z = 3
    z = 4
```

### System information

```shell
- Ubuntu (WSL 2)
- mojo 0.5.0 (6e50a738)
- modular 0.2.2 (95d42445)
- VSCode 1.84.2
- Mojo 🔥 extension v0.5.0
```

**Comment by lattner** [comment]
@River707 VSCode not running check lifetimes ?

**Comment by River707** [comment]
This should be fixed in 0.6!
